### PR TITLE
Feat(eos_cli_config_gen): FlexRoute - add support for TCAM assisted ACLs #1656

### LINKED ISSUE #1656: Feat(rs_eos_cli_config_gen): FlexRoute - add support for TCAM assisted ACLs
### Enhancement summary

From https://aristanetworks.force.com/AristaCommunity/s/article/inet-edge-config

Another optimization we recommend for edge deployments utilizing FlexRoute is to change the internal mechanism the switch router will use to implement Access-Control Lists.  The hardware typically utilized in high-scale edge routing deployments supports Arista AlgoMatch technology, which is enabled by default.  However, to ensure proper allocation of system resources, it is preferable at this time to change the implementation to leverage the system TCAM for ACLs. 

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Internet edge routing

### Describe the solution you would like

```yaml
hardware:
  access_list:
    mechanism: < algomatch | none | tcam | default -> algomatch >
```

### Describe alternatives you have considered

Using a configlet

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by c-po**
## Change Summary

From https://aristanetworks.force.com/AristaCommunity/s/article/inet-edge-config

Another optimization we recommend for edge deployments utilizing FlexRoute is to change the internal mechanism the switch router will use to implement Access-Control Lists. The hardware typically utilized in high-scale edge routing deployments supports Arista AlgoMatch technology, which is enabled by default. However, to ensure proper allocation of system resources, it is preferable at this time to change the implementation to leverage the system TCAM for ACLs.

## Related Issue(s)

Fixes #1656

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

```yaml
hardware:
  access_list:
    mechanism: < algomatch | none | tcam | default -> algomatch >
```

## How to test

Moelcule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: the default and supported values in EOS is highly platform dependent, so I would prefer not to have any default in `eos_cli_config_gen`. Just configure whatever value the user specifies, or nothing. This is already what you do in the template, so just remove the default from the doc.

**Review by ClausHolbechArista — APPROVED** [review]
I don't think we need docs for this feature, so LGTM

**Review by emilarista — APPROVED** [review]
LGTM once doc is fixed according to Claus' comment.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md
Comment (c-po) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Done

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Pull request merged**
